apps/arweave/src/ar_storage.erl

```diff
@@ -299,28 +299,30 @@ put_tx_confirmation_data(B) ->
 %% @doc Return {BlockHeight, BlockHash} belonging to the block where
 %% the given transaction was included.
 get_tx_confirmation_data(TXID) ->
-	case ar_kv:get(tx_confirmation_db, TXID) of
-		{ok, Binary} ->
-			{ok, binary_to_term(Binary)};
-		not_found ->
-			{ok, Config} = application:get_env(arweave, config),
-			case lists:member(arql, Config#config.disable) of
-				true ->
-					not_found;
-				_ ->
-					case catch ar_arql_db:select_block_by_tx_id(ar_util:encode(TXID)) of
-						{ok, #{
-							height := Height,
-							indep_hash := EncodedIndepHash
-						}} ->
-							{ok, {Height, ar_util:decode(EncodedIndepHash)}};
-						not_found ->
-							not_found;
-						{'EXIT', {timeout, {gen_server, call, [ar_arql_db, _]}}} ->
-							{error, timeout}
-					end
-			end
-	end.
+    case ar_kv:get(tx_confirmation_db, TXID) of
+        {ok, Binary} ->
+            {ok, binary_to_term(Binary)};
+        not_found ->
+            {ok, Config} = application:get_env(arweave, config),
+            case lists:member(arql, Config#config.disable) of
+                true ->
+                    {error, not_found};
```

Comment:
I think this will break stuff. If there's a rule forcing that all function return values match `{error, XXX}` or `{ok, XXX}` we should disable it.

If we really want to we can enable it later, but it will have wide ranging impact on logic so it should be its own PR

Reply:
this isn't Elvis, it was my hands getting dirty on cleaning the code where it was not single case statement, I'll revert

Reply:
it's reverted, now it's just not_found and